### DISCUSSION TITLE
Is it possible to disable workspace auth?

### DISCUSSION THREAD

**Discussion by gskolber** [coder/coder#16745] [Workspace Provisioning & Templates] [answered]
#### **Summary**:
We are using Coder to **provision VSCode instances for programming tests**, and we need a way to disable workspace authentication in these environments. While we understand this may not be a current priority, we’d like to know if there’s a native solution for this, as we haven’t found anything in the documentation. We've already tried several approaches, including passing the `coder_session_token` via an iframe, but the session doesn’t seem to persist.

---

#### **Our Use Case**:

We are using Coder to provision environments for programming tests. These are isolated, temporary environments specifically created for test-takers. In these cases, authentication becomes a blocker, as the participants don’t need to go through the standard authentication process.

- **Programming Tests**: Since these environments are short-lived and dedicated to candidates taking tests, requiring full authentication for each session is unnecessary and slows down the test setup process.
- **Non-Production Use**: The environments contain no sensitive data or production-level activities, making authentication more of a friction point rather than a security requirement.

---

#### **What We Have Tried**:

1. **Passing `coder_session_token` via iframe**:
   - We attempted to pass the `coder_session_token` as a query parameter in the iframe to persist the session, but it appears that the session does not persist when loading the environment via an iframe.

2. **Reverse Proxy-Based Bypass Using NGINX**:
   - We configured NGINX to intercept requests such as `/api/v2/users/me` and `/api/v2/authcheck` and return mock responses for `401 Unauthorized` errors.
   - While this worked, it introduced unnecessary complexity, required ongoing maintenance, and wasn’t sustainable for scaling the number of environments.

3. **Manual Adjustments to Authentication Logic**:
   - We explored manually bypassing authentication within the API, but this requires too many manual changes and is not ideal for ongoing use.

---

#### **Proposed Solution**:

We’re wondering if there’s a **native solution** for bypassing authentication in these specific use cases, such as a configurable flag or environment variable. Something like:

- **ENV**: `DISABLE_WORKSPACE_AUTH=true`
- **YAML Config**: `disableWorkspaceAuth: true`

This would allow us to:
- Bypass authentication for programming test environments.
- Avoid unnecessary complexity with proxy configurations or manual API adjustments.
- Simplify the workflow for participants who are only focused on completing their tests.

---

#### **Note**:
We understand that implementing this feature might not be a priority at this time. However, it would be helpful to know if there’s already an existing way to achieve this, as we couldn’t find anything in the documentation.

**Comment by stirby** [comment]
Would [port sharing](https://coder.com/docs/admin/networking/port-forwarding#sharing-ports) solve you use case? For example, you could install the [`code-server` module](https://registry.coder.com/modules/code-server) and share the port it runs on (13337 by default) at the `public` level, then anyone could access the app.

![Image](https://github.com/user-attachments/assets/7ebdb171-03b5-46cc-bd86-5fd6fed46472)

**Comment by gskolber** [comment]
We only set share as public. Easy!

Thanks for the attention

![Image](https://github.com/user-attachments/assets/deafe7d1-c520-4ca8-9356-ec298547d47f)

**Comment by gskolber** [comment, marked as answer]
```terraform
module "code-server" {
  source         = "./code-server"
  agent_id       = coder_agent.main.id
  script_name    = data.coder_parameter.script_name.value
  attempt_id     = data.coder_parameter.attempt_id.value
  share          = "public"
}
```

**Reply by matifali** [reply]
Please note that this will make the app accessible to anyone with the link.
